### PR TITLE
Fix/various

### PR DESCRIPTION
**Pull request by tjallingt**

### CODE REVIEW

File: lib/App.js [hunk 1]
```diff
@@ -1444,7 +1442,7 @@ class App {
         appJson = appJson.toString();
         appJson = JSON.parse(appJson);
         appJson.drivers = appJson.drivers || [];
-        appJson.drivers.push( driverJson );
+        appJson.drivers.push({ id: driverId, ...driverJson });
```
Comment (mrlukasbos) [inline lib/App.js:1445]: Was the ID previously not set?

File: lib/App.js [hunk 2]
```diff
@@ -1820,21 +1817,17 @@ class App {
   }
 
   async createFlow() {
-    let flowJson = await this.createFlowJson();
-
-    if( !flowJson ) throw new Error('Could not create valid Flow');
```
Comment (mrlukasbos) [inline lib/App.js:1825]: Why not throw this error?

Comment (tjallingt) [reply lib/App.js:1825]: cause the only way this returns undefined is if the user chose to skip, so we don't really need to show an error in that case

File: lib/App.js (hunk 1 above)
Comment (tjallingt) [reply lib/App.js:1445]: no its the opposite, I removed the id from `driverJson` but in this case (when homeycompose is not used) we still need it so I add it back into the object

Comment (mrlukasbos) [reply lib/App.js:1445]: Ah i see.

File: lib/App.js [hunk 3]
```diff
@@ -1639,35 +1634,37 @@ class App {
       ? await App._questionFlowTokens()
       : [];
 
-    const confirm = await inquirer.prompt([
+    const { confirm } = await inquirer.prompt([
```
Comment (mrlukasbos) [inline lib/App.js:1637]: Nice catch

**Review by mrlukasbos — APPROVED** [review]

**Pull request merged**